aws/data_source_aws_elb_test.go

```diff
@@ -0,0 +1,113 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccDataSourceAWSELB_basic(t *testing.T) {
+	lbName := fmt.Sprintf("testaccawselb-basic-%s", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAWSELBConfigBasic(lbName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.aws_elb.elb_test", "name", lbName),
+					resource.TestCheckResourceAttr("data.aws_elb.elb_test", "cross_zone_load_balancing", "true"),
+					resource.TestCheckResourceAttr("data.aws_elb.elb_test", "idle_timeout", "30"),
+					resource.TestCheckResourceAttr("data.aws_elb.elb_test", "internal", "true"),
+					resource.TestCheckResourceAttr("data.aws_elb.elb_test", "subnets.#", "2"),
+					resource.TestCheckResourceAttr("data.aws_elb.elb_test", "security_groups.#", "1"),
+					resource.TestCheckResourceAttr("data.aws_elb.elb_test", "tags.%", "1"),
+					resource.TestCheckResourceAttr("data.aws_elb.elb_test", "tags.TestName", "TestAccAWSELB_basic"),
+					resource.TestCheckResourceAttrSet("data.aws_elb.elb_test", "dns_name"),
+					resource.TestCheckResourceAttrSet("data.aws_elb.elb_test", "zone_id"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceAWSELBConfigBasic(lbName string) string {
+	return fmt.Sprintf(`resource "aws_elb" "elb_test" {
+  name            = "%s"
+  internal        = true
+  security_groups = ["${aws_security_group.elb_test.id}"]
+  subnets         = ["${aws_subnet.elb_test.*.id}"]
+
+  idle_timeout = 30
+  enable_deletion_protection = false
```

Comment:
This is an ALB/NLB specific field which needs to be removed. It's also the reason why the test is currently failing:

```
=== RUN   TestAccDataSourceAWSELB_basic
--- FAIL: TestAccDataSourceAWSELB_basic (3.64s)
	testing.go:503: Step 0 error: Configuration is invalid.

		Warnings: []string(nil)

		Errors: []string{"aws_elb.elb_test: : invalid or unknown key: enable_deletion_protection"}
```